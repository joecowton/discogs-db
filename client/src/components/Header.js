// @flow
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { css } from 'glamor';

type Props = {
    auth: string,
};

const style = css({
    navbar: {
        display: 'grid',
        gridTemplateColumns: 'repeat(5, 1fr)',
        gridAutoRows: 'auto',
        gridGap: '1em',
        alignItems: 'center',
        textAlign: 'center',
        borderBottom: '4px solid black',
        background: 'green',
    },
});

export class Header extends Component<Props> {
    render() {
        return (
            <div>
                <nav className="navbar navbar-inverse" {...style}>
                    <div className="container-fluid">
                        <div className="navbar-logo">
                            <Link to={this.props.auth ? '/releases/all' : '/'}>
                                Main
                            </Link>
                            <ul className="right">{this.renderContent()}</ul>
                        </div>
                    </div>
                </nav>
            </div>
        );
    }

    renderContent() {
        switch (this.props.auth) {
            case null:
                return;
            case false:
                return (
                    <li>
                        <a href="/auth/google">Login</a>
                    </li>
                );
            default:
                return [
                    <div key="1">
                        <li key="1">
                            <Link to={this.props.auth ? '/releases/new' : '/'}>
                                Add New
                            </Link>
                        </li>
                        <li key="2">
                            <a href="/api/logout">Logout</a>
                        </li>
                    </div>,
                ];
        }
    }
}

function mapStateToProps({ auth }) {
    return { auth };
}

export default connect(mapStateToProps)(Header);
