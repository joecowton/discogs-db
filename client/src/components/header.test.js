import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

const defaultProps = {
    auth: true,
};

describe('Header', () => {
    it('should render the component', () => {
        const wrapper = shallow(<Header {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
