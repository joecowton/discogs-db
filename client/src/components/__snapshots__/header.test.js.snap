// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the component 1`] = `
<div>
  <nav
    className="navbar navbar-inverse"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-logo"
      >
        <Link
          replace={false}
          to="/releases/all"
        >
          Main
        </Link>
        <ul
          className="right"
        >
          <div
            key="1"
          >
            <li
              key="1"
            >
              <Link
                replace={false}
                to="/releases/new"
              >
                Add New
              </Link>
            </li>
            <li
              key="2"
            >
              <a
                href="/api/logout"
              >
                Logout
              </a>
            </li>
          </div>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
